packages/headless/package.json

```diff
@@ -48,12 +48,14 @@
   "dependencies": {
     "@coveo/bueno": "0.43.7",
     "@reduxjs/toolkit": "1.8.5",
+    "@types/event-source-polyfill": "^1.0.1",
     "@types/pino": "6.3.12",
     "@types/redux-mock-store": "1.0.3",
     "abab": "2.0.6",
-    "coveo.analytics": "2.28.5",
+    "coveo.analytics": "2.28.7",
     "cross-fetch": "3.1.5",
     "dayjs": "1.11.5",
+    "event-source-polyfill": "^1.0.31",
```

Comment:
How much thoughts/analysis have we put into checking how the polyfill works, it's size, who's the developer, when and why it's needed, etc.

Any dependency we add here has to be very solid.

It seems that https://caniuse.com/eventsource is supported by all browsers we care about ?
And so we'd only need it for Node JS (or do we ?)

Reply:
This is indeed my least favorite part of this project. EventSource is supported by all browsers but the reason we need something else is that it is extremely simple and limited as it doesn't even allow us to specify an authorization header. The auth check is a requirement we have so already we need some type of polyfill that expands the base features.

The lib is the best I could find after some research. the class we import is this one https://github.com/Yaffle/EventSource/blob/master/src/eventsource.js which is 33kb. The library has a decent amount of usage and 30-ish contributors but I don't know what our standards are normally.

Writing our own version of an SSE client was off the table for this beta feature but could probably be done later if that's what we need.

What do you think?

Reply:
Is this something we should discuss with the team in charge of the API then, and arrange for a different way to negotiate authentication ? Possibly using a `auth` event to send the token ?

Clients not using Headless will face the same problem, and so we're forcing a polyfill on everyone, because it does not follow "normal web standards for event source".

Is @AndyCoveo team's in charge of this service ? What is your opinion on this one ?

Reply:
Could also look at full blown web socket 😁

Reply:
I had a live discussion with Andy.
I'll talk with security also.

What it boils down to is that I find it weird that we'll have in production an API that is essentially not usable with the standard as it is defined on MDN/what browser implements.

Which polyfill we end up using if that's really the only course of action is a secondary concern.

Reply:
I'm strongly against including the polyfill. Details in Slack.

Reply:
Replaced the polyfill with https://www.npmjs.com/package/@microsoft/fetch-event-source